code/game/objects/effects/landmarks.dm

```diff
@@ -588,6 +588,60 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/awaystart) //Without this away mission
 	ruin_template = null
 	. = ..()
 
+
+/// Mob spawners, spawns a mob and deletes the landmark
+
+/obj/effect/landmark/mob_spawner
+	///The mob we use for the spawner
+	var/mobtype = null
+	icon = 'icons/effects/spawner_icons.dmi'
+
+/obj/effect/landmark/mob_spawner/Initialize(mapload)
+	. = ..() 
+	new mobtype(loc)
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/landmark/mob_spawner/goliath
+	mobtype = /mob/living/simple_animal/hostile/asteroid/goliath/beast
+	icon_state = "questionmark"
+
+/obj/effect/landmark/mob_spawner/goliath/Initialize(mapload)
+	if(prob(1))
+		new /mob/living/simple_animal/hostile/asteroid/goliath/beast/ancient(loc)
+		return INITIALIZE_HINT_QDEL
```

Comment:
for these `new`s, why not just set the `mobtype` equal to this path here and then let the parent function do all the heavy lifting? Then you wouldn't have to duplicate `return INITIALIZE_HINT_QDEL` which would cut down on code usage

Reply:
Damn you're right, lemme fix that in the next commit

Reply:
Suggestion added